README.md

```diff
@@ -981,9 +982,60 @@ Optional configuration:
         onFailure(error);
      }
     ```
-## OpenId Connect
 
-Options for the [OpenId Connect](http://developer.okta.com/docs/api/resources/oidc.html) authentication flow. This flow is required for social authentication, and requires OAuth client registration with Okta. For instructions, see [Social Authentication](http://developer.okta.com/docs/api/resources/social_authentication.html).
+## IdP Discovery
+**:information_source: EA feature:** The Identity Provider (IdP) Discovery feature is currently an [EA feature](https://developer.okta.com/docs/api/getting_started/releases-at-okta#early-access-ea).
+
+IdP Discovery enables you to route users to different 3rd Party IdPs that are connected to your Okta Org. Users can federate back into the primary org after authenticating at the IdP.
+
+To use IdP Discovery in your application, first ensure that the `IDP_DISCOVERY` feature flag is enabled for your Org and configure an identity provider routing policy in the Okta admin panel.
+Then, in the widget configuration, set `features.idpDiscovery` to `true` and add additional configs under the `idpDiscovery` key on the [`OktaSignIn`](#new-oktasigninconfig) object.
+
+```javascript
+var signIn = new OktaSignIn({
+  baseUrl: 'https://your-org.okta.com',
+  ... ...
+  idpDiscovery: {
+    requestContext: '/a/app/request/context',
+  },
+  features: {
+    idpDiscovery: true
+  }
+});
+signIn.renderEl(
+  {...},
+  function (res) {
+    if (res.status === 'IDP_DISCOVERY') {
+      res.idpDiscovery.redirectToIdp('/a/app/request/context');
+      return;
+    }
+  }
+);
+```
+
+The IdP Discovery authentication flow in widget will be
+
+1. If a routing policy with a username/domain condition is configured, the widget will enter identifier first flow
+2. Otherwise, the widget will enter primary authentication flow.
+
+For the identifier first flow,
+
+1. The widget will display an identifier first page for the user to enter an Okta userName to determine the IdP to be used for authentication.
+2. If the IdP is your Okta org, the widget will transition to the primary authentication flow.
+3. If the IdP is a 3rd party IdP or a different Okta org, the widget will invoke the [success callback](#rendereloptions-success-error) with `response.status` as `IDP_DISCOVERY`.
+
+### Additional configuration
+
+- **idpDiscovery.requestContext**: a context of that which the user is trying to access, such as the path of an app
+
+### Additions in the success callback
+
+- `response.status` is `IDP_DISCOVERY` when the authentication needs to be done agaist 3rd party IdP.
+- `res.idpDiscovery.redirectToIdp` is a function that is used for redirecting to relative path of the 3rd party IdP. It **has to** be same as **idpDiscovery.requestContext**.
```

Comment:
You mean the parameter has to be same .... ?

Reply:
oops.. yes!